BuildResidentialHPXML/measure.rb

```diff
@@ -0,0 +1,3573 @@
+    arg.setDescription("The control type of the HVAC system.")
+    arg.setDefaultValue("manual thermostat")
+    args << arg
+
+    arg = OpenStudio::Measure::OSArgument::makeDoubleArgument("heating_setpoint_temp", true)
+    arg.setDisplayName("Heating Setpoint Temperature")
+    arg.setDescription("Specify the heating setpoint temperature.")
+    arg.setUnits("degrees F")
+    arg.setDefaultValue(71)
+    args << arg
+
+    arg = OpenStudio::Measure::OSArgument::makeDoubleArgument("heating_setback_temp", true)
+    arg.setDisplayName("Heating Setback Temperature")
+    arg.setDescription("Specify the heating setback temperature.")
+    arg.setUnits("degrees F")
+    arg.setDefaultValue(71)
+    args << arg
+
+    arg = OpenStudio::Measure::OSArgument::makeDoubleArgument("heating_setback_hours_per_week", true)
+    arg.setDisplayName("Heating Setback Hours per Week")
+    arg.setDescription("Specify the heating setback number of hours per week value.")
+    arg.setDefaultValue(0)
+    args << arg
+
+    arg = OpenStudio::Measure::OSArgument::makeDoubleArgument("heating_setback_start_hour", true)
+    arg.setDisplayName("Heating Setback Start Hour")
+    arg.setDescription("Specify the heating setback start hour value. 0 = midnight, 12 = noon")
+    arg.setDefaultValue(0)
+    args << arg
+
+    arg = OpenStudio::Measure::OSArgument::makeDoubleArgument("cooling_setpoint_temp", true)
+    arg.setDisplayName("Cooling Setpoint Temperature")
+    arg.setDescription("Specify the cooling setpoint temperature.")
+    arg.setUnits("degrees F")
+    arg.setDefaultValue(76)
+    args << arg
+
+    arg = OpenStudio::Measure::OSArgument::makeDoubleArgument("cooling_setup_temp", true)
+    arg.setDisplayName("Cooling Setup Temperature")
+    arg.setDescription("Specify the cooling setup temperature.")
+    arg.setUnits("degrees F")
+    arg.setDefaultValue(76)
+    args << arg
+
+    arg = OpenStudio::Measure::OSArgument::makeDoubleArgument("cooling_setup_hours_per_week", true)
+    arg.setDisplayName("Cooling Setup Hours per Week")
+    arg.setDescription("Specify the cooling setup number of hours per week value.")
+    arg.setDefaultValue(0)
+    args << arg
+
+    arg = OpenStudio::Measure::OSArgument::makeDoubleArgument("cooling_setup_start_hour", true)
+    arg.setDisplayName("Cooling Setup Start Hour")
+    arg.setDescription("Specify the cooling setup start hour value. 0 = midnight, 12 = noon")
+    arg.setDefaultValue(0)
+    args << arg
+
+    duct_leakage_units_choices = OpenStudio::StringVector.new
+    duct_leakage_units_choices << "CFM25"
+    duct_leakage_units_choices << "Percent"
+
+    duct_location_choices = OpenStudio::StringVector.new
+    duct_location_choices << Constants.Auto
+    duct_location_choices << "living space"
+    duct_location_choices << "attic - vented"
+    duct_location_choices << "attic - unvented"
+    duct_location_choices << "crawlspace - vented"
+    duct_location_choices << "outside"
```

Comment:
There are choices missing here like "garage", "crawlspace - unvented", ...

Reply:
Any others that are missing? (I don't think I saw any test files that have ducts in these missing locations...)

Reply:
Stop looking at the test files and look at the [EPvalidator specification](https://github.com/NREL/OpenStudio-HPXML/blob/master/HPXMLtoOpenStudio/resources/EPvalidator.rb#L437) (here and for other elements with enumerations).